### PR TITLE
Support manually setting the dtype of the model and tuned lens weights

### PR DESCRIPTION
**Pull request by norabelrose**
Importantly this PR also changes the dtype of tuned lens parameters. Before they were always float32 (or more precisely, they were always whatever is returned by `torch.get_default_dtype()`. But that's stupid and wastes VRAM for large models. We now make the tuned lens the same dtype as the model parameters (except in the case of int8, where we use float16).

This PR also rolls the `--int8` functionality into the new `--precision` flag; just set `--precision int8`

As a bonus, this PR also ensures that HuggingFace doesn't stupidly try to load a model from a tuned lens directory with the same name using the new `prevent_name_conflicts` context manager.

### CODE REVIEW

File: tuned_lens/scripts/ingredients.py [hunk 1]
```diff
@@ -91,8 +92,8 @@ class Model:
     name: str
     """Name of model to use in the Huggingface Hub."""
 
-    int8: bool = field(action="store_true", default=False)
-    """Load the model weights in mixed int8 mode with `bitsandbytes`."""
+    precision: Literal["auto", "bfloat16", "float16", "float32", "int8"] = "auto"
```
Comment (levmckinney) [inline tuned_lens/scripts/ingredients.py:95]: I don't think simple parsing currently supports Literals. Can we make this an enum.

Comment (levmckinney) [reply tuned_lens/scripts/ingredients.py:95]: I guess it does now.

Comment (norabelrose) [reply tuned_lens/scripts/ingredients.py:95]: I caused simple parsing to support them as of v0.1.1

File: tuned_lens/utils.py [hunk 2]
```diff
@@ -111,6 +114,18 @@ def pairwise(it: Iterable[T]) -> Iterable[tuple[T, T]]:
     yield from zip(it, islice(it, 1, None))
 
 
+@contextmanager
+def prevent_name_conflicts():
```
[the remark this answers is not in the document]
Comment (norabelrose) [reply tuned_lens/utils.py]: fixed

**Review by levmckinney — APPROVED** [review]
LGTM

**Pull request merged**